File: monitorlibrary/lib/data/photo.dart
import 'package:flutter/material.dart';
import 'package:monitorlibrary/data/position.dart';

class Photo {
  String url, thumbnailUrl, caption, created, photoId;
  String userId, organizationId;
  String userName;
  Position projectPosition;
  double distanceFromProjectPosition;
  String projectId, projectName;
  int height, width;

  Photo(
      {@required this.url,
      @required this.caption,
      @required this.created,
      @required this.userId,
      @required this.userName,
      @required this.projectPosition,
      @required this.distanceFromProjectPosition,
      @required this.projectId,
      @required this.thumbnailUrl,
      @required this.photoId,
      @required this.organizationId,
      @required this.projectName,
      @required this.height,
      @required this.width});

  Photo.fromJson(Map data) {
    this.url = data['url'];
    this.thumbnailUrl = data['thumbnailUrl'];
    this.caption = data['caption'];
    this.height = data['height'];
    this.width = data['width'];
    this.created = data['created'];
    this.organizationId = data['organizationId'];
    this.userId = data['userId'];
    this.photoId = data['photoId'];
    this.userName = data['userName'];
    this.distanceFromProjectPosition = data['distanceFromProjectPosition'];
    this.projectId = data['projectId'];
    this.projectName = data['projectName'];
    if (data['projectPosition'] != null) {
      this.projectPosition = Position.fromJson(data['projectPosition']);
    }
    if (this.height == null) {
      this.height = -5;
    }
    if (this.width == null) {
      this.width = -10;
    }
  }
  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {
      'url': url,
      'caption': caption,
      'created': created,
      'width': width,
      'height': height,
      'userId': userId,
      'organizationId': organizationId,
      'photoId': photoId,
      'userName': userName,
      'distanceFromProjectPosition': distanceFromProjectPosition,
      'projectId': projectId,
      'projectName': projectName,
      'thumbnailUrl': thumbnailUrl,
      'projectPosition':
          projectPosition == null ? null : projectPosition.toJson()
    };
    return map;
  }
}

class Video {
  String url, caption, created, thumbnailUrl, videoId;
  String userId, userName, organizationId;
  Position projectPosition;
  double distanceFromProjectPosition;
  String projectId, projectName;

  Video(
      {@required
          this.url,
      this.caption,
      @required
          this.created,
      @required
          this.userId,
      @required
          this.userName,
      @required
          this.projectPosition,
      @required
          this.distanceFromProjectPosition,
      @required
          this.projectId,
      @required
          this.thumbnailUrl,
      @required
          this.videoId,
      @required
          this.organizationId,
      @required
          this.projectName}); // Video({@required this.url, this.userId, @required this.created});

  Video.fromJson(Map data) {
    this.url = data['url'];
    this.caption = data['caption'];
    this.created = data['created'];
    this.userId = data['userId'];
    this.organizationId = data['organizationId'];
    this.thumbnailUrl = data['thumbnailUrl'];
    this.videoId = data['videoId'];
    this.userName = data['userName'];
    this.distanceFromProjectPosition = data['distanceFromProjectPosition'];
    this.projectId = data['projectId'];
    this.projectName = data['projectName'];
    if (data['projectPosition'] != null) {
      this.projectPosition = Position.fromJson(data['projectPosition']);
    }
  }
  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {
      'url': url,
      'caption': caption,
      'created': created,
      'userId': userId,
      'videoId': videoId,
      'organizationId': organizationId,
      'userName': userName,
      'thumbnailUrl': thumbnailUrl,
      'distanceFromProjectPosition': distanceFromProjectPosition,
      'projectId': projectId,
      'projectName': projectName,
      'projectPosition':
          projectPosition == null ? null : projectPosition.toJson()
    };
    return map;
  }
}

class Condition {
  String url, caption, created;
  String userId, userName;
  Position projectPosition;
  int rating;
  String projectId, projectName;

  Condition(
      {@required
          this.url,
      this.caption,
      @required
          this.created,
      @required
          this.userId,
      @required
          this.userName,
      @required
          this.projectPosition,
      @required
          this.rating,
      @required
          this.projectId,
      @required
          this.projectName}); // Video({@required this.url, this.userId, @required this.created});

  Condition.fromJson(Map data) {
    this.url = data['url'];
    this.caption = data['caption'];
    this.created = data['created'];
    this.userId = data['userId'];

    this.userName = data['userName'];
    this.rating = data['rating'];
    this.projectId = data['projectId'];
    this.projectName = data['projectName'];
    if (data['projectPosition'] != null) {
      this.projectPosition = Position.fromJson(data['projectPosition']);
    }
  }
  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {
      'url': url,
      'caption': caption,
      'created': created,
      'userId': userId,
      'userName': userName,
      'rating': rating,
      'projectId': projectId,
      'projectName': projectName,
      'projectPosition':
          projectPosition == null ? null : projectPosition.toJson()
    };
    return map;
  }
}

/*
data class Condition(var _partitionKey: String?, @Id var _id: String?,
                 var projectId: String,
                 var projectName: String,
                 var rating: Int,
                 var caption: String?,
                 var userId: String,
                 var userName: String,
                 var created: String) {}
 */

File: backend/src/main/java/com/monitor/backend/data/Photo.java
package com.monitor.backend.data;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


@Document(collection = "photo")
public class Photo {
    private String _partitionKey;
    @Id
    private String _id;
    private String projectId, projectPositionId;
    private String projectName;
    private String photoId;
    private String organizationId;
    private Position projectPosition;
    private double distanceFromProjectPosition;
    private String url;
    private String thumbnailUrl;
    private String caption;
    private String userId;
    private String userName;
    private String created;
    private int height, width;

    public Photo() {
    }

    public Photo(String _partitionKey, String _id, String projectId, String projectPositionId, String projectName, String photoId, String organizationId, Position projectPosition, double distanceFromProjectPosition, String url, String thumbnailUrl, String caption, String userId, String created, int height) {
        this._partitionKey = _partitionKey;
        this._id = _id;
        this.projectId = projectId;
        this.projectPositionId = projectPositionId;
        this.projectName = projectName;
        this.photoId = photoId;
        this.organizationId = organizationId;
        this.projectPosition = projectPosition;
        this.distanceFromProjectPosition = distanceFromProjectPosition;
        this.url = url;
        this.thumbnailUrl = thumbnailUrl;
        this.caption = caption;
        this.userId = userId;
        this.created = created;
        this.height = height;
    }

    public String getProjectPositionId() {
        return projectPositionId;
    }

    public void setProjectPositionId(String projectPositionId) {
        this.projectPositionId = projectPositionId;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public String get_partitionKey() {
        return _partitionKey;
    }

    public void set_partitionKey(String _partitionKey) {
        this._partitionKey = _partitionKey;
    }

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getProjectId() {
        return projectId;
    }

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public String getPhotoId() {
        return photoId;
    }

    public void setPhotoId(String photoId) {
        this.photoId = photoId;
    }

    public String getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(String organizationId) {
        this.organizationId = organizationId;
    }

    public Position getProjectPosition() {
        return projectPosition;
    }

    public void setProjectPosition(Position projectPosition) {
        this.projectPosition = projectPosition;
    }

    public double getDistanceFromProjectPosition() {
        return distanceFromProjectPosition;
    }

    public void setDistanceFromProjectPosition(double distanceFromProjectPosition) {
        this.distanceFromProjectPosition = distanceFromProjectPosition;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getThumbnailUrl() {
        return thumbnailUrl;
    }

    public void setThumbnailUrl(String thumbnailUrl) {
        this.thumbnailUrl = thumbnailUrl;
    }

    public String getCaption() {
        return caption;
    }

    public void setCaption(String caption) {
        this.caption = caption;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }
}
